torch/csrc/jit/mobile/nnc/context.cpp

```diff
@@ -157,9 +157,18 @@ void Function::init_execution_state() const {
   // Keep empty slots to fill in inputs/outputs pointers at execution time.
   arguments.resize(input_args + output_args);
 
-  // Fill in parameter pointers.
+  // Fill in parameters as untyped raw pointers.
+  // The underlying storage of the parameters should be owned by `parameters_`,
+  // which should be alive when `execution_state_` is being used.
   for (const auto& param : parameters_) {
-    arguments.emplace_back(param.data_ptr());
+    const c10::IValue& ivalue = (c10::IValue)param;
+    if (ivalue.isTensor()) {
+      arguments.emplace_back(ivalue.toTensor().data_ptr());
+    } else if (ivalue.isObject()) {
```

Comment:
Do you mean to check if this is a custom class? Because I assume `isObject` can return true for other torchscript classes which were defined in python.

Reply:
I initially looked at ivalue.isCustomClass() method, which requires the concrete custom class type as template parameter, so I didn't use it.
Found a helper method in torch::isCustomClass() that can do the check without providing the concrete type here.